Add -test results to the panel

yzgyyang opened the pull request:
Add results for -test builds when available.

When I was there:
- Added global variables for local and staging env
- Added a JSON example with a server for local dev

yzgyyang committed:
Add -test results to the panel

yzgyyang commented:
Preview at https://people.freebsd.org/~ygy/tinderbox/.

lwhsu commented:
The result looks good to me, maybe we can arrange the information as following:
```
Revision: rXXXX
Build: <date> <details>
Test: <date> <results>
```
And use text color on <date> field to mark the result is successful, unstable or failed.

One thing I overlooked when talking about the idea, currently I guess the shown test result is from the last or last completed test build, but those two builds might not have directly relationship, i.e., they are not against the same revision.

The "started by" information can be retrieved from the item with `hudson.model.CauseAction` class in `actions` array in the build information.
See https://ci.freebsd.org/job/FreeBSD-head-amd64-test/17320/api/json?pretty=true for example, it should also be retrieved from https://ci.freebsd.org/api/json by providing right parameters.

This might increase much complexity of the implementation, if it is too difficult, let's find a better way to organize and display those data.

lwhsu commented:
example.json looks like sample data, does it need to be included in the final commit?
Also what's the reason of renaming server.js to server-proxy.js?

yzgyyang committed:
Use side-by-side layout

While I was there, also:
- Made the table header sticky (on modern browsers)
- Only show last successful build for jobs with FAILURE

yzgyyang committed:
Add a second level of header and cleanup code

yzgyyang commented:
> One thing I overlooked when talking about the idea, currently I guess the shown test result is from the last or last completed test build, but those two builds might not have a direct relationship, i.e., they are not against the same revision.

@lwhsu That makes sense. I have implemented a side-by-side view for build and test, as well as adding a sticky header for the table. The end result looks like this: (https://people.freebsd.org/~ygy/tinderbox/)

![image](https://user-images.githubusercontent.com/16748524/101146207-b0b98380-3655-11eb-84b8-a6d255fab777.png)

Please let me know how I can improve the display to be clearer with sufficient info.

> Also what's the reason of renaming server.js to server-proxy.js?
> example.json looks like sample data, does it need to be included in the final commit?

`proxy-server.js` is for forwarding the result without the CORS restriction, and since I don't have a stable connection, I made a helper `json-server.js` that serves an `example.json` directly on localhost. IMO these two server scripts are just for development, so I could possibly move them to a `dev` folder to avoid confusion? (or is the server actually using the `server.js`?)
